paddle/platform/transform.h

```diff
@@ -31,13 +31,14 @@ namespace paddle {
 namespace platform {
 // Transform on host or device. It provides the same API in std library.
 template <typename InputIter, typename OutputIter, typename UnaryOperation>
-void Transform(const DeviceContext& context, InputIter first, InputIter last,
-               OutputIter result, UnaryOperation op) {
+inline static void Transform(const DeviceContext& context, InputIter first,
```

Comment:
Since we use `static`, the `Transform` will not generate symbols. `cc` or `cu` will use different `Transform` method.

Reply:
Thought it was the `inline` keyword that makes `Transform` not generating symbols?

Reply:
I do not think this solution is better than #4201. There are two problems with this solution:
1. The `Transform` interface in .cu.o has a useless cpu code branch.
2. If there are many .cc and .cu are used `Transform` interface, will compile a lot of duplicate code.

Reply:
Maybe we should force inline? I think using a functor is a little bit complex.

Reply:
1. We may invoke CPU Transform in `.cu` file. So that branch is not useless. Also, if we make `place` as template parameter, we should use `PADDLE_ENFORCE` to check the template place is same as the `context.GetPlace()`. So the `if` branch will not be removed no matter how we implement `Transform`.

2. In fact, the `BinaryFunctor`, `UnaryFunctor` and the `Iterator` are template parameter. Each tiime we invoke `Transform`, a new code will be generated. No matter how we implement it.

To make `Transform` a simple method, not a templated functor, can make users invoke it easily.

It is easier for user write `Transform(ctx, input, input+size, output, Functor);` than `Transform<Place>()(ctx, input, input+size, output, Functor);`.